src/templates/infra/api/rest/index.ejs

```diff
@@ -1,27 +1,55 @@
-const { json } = require('body-parser')
-const { generateRoutes } = require('@herbsjs/herbs2rest')
 const cors = require('cors')
 const express = require('express')
-const renderShelfHTML = require('@herbsjs/herbsshelf')
-const usecases = require('../../../domain/usecases')
+const { json } = require('body-parser')
+const camelCase = require('lodash.camelcase')
 const db = require('../../../infra/data/database')
+const usecases = require('../../../domain/usecases')
+const renderShelfHTML = require('@herbsjs/herbsshelf')
+const { generateRoutes } = require('@herbsjs/herbs2rest')
 const repositoriesFactory = require('../../../infra/data/repositories')
 
-function prepareUsecases(usecases){
+function cloneUsecases(usecases){
     return Promise.all(usecases.map(uc => {
         const clonedUC = { ...uc}
         clonedUC.usecase = clonedUC.usecase({})()
         return clonedUC
     }))
 }
 
+const newRoute = (usecase, useId) => { 
+    const route = { usecase }
+    if(useId) route.id = 'id'
+    return route
+}
+
 async function prepareRoutes(config){
     const conn = await db.factory(config) 
     const repositories = await repositoriesFactory(conn)
 
-    const routes = <%- props.routes %>
+    // groupBy group
+    const ucByGroup = usecases.reduce((acc, obj) => {
```

Comment:
I believe this code block is super complex to "user/developer" mangage,  it could be into [@herbs/herbs2rest](https://www.npmjs.com/package/@herbsjs/herbs2rest) in the future.
After this PR to be merged I will open an issue into Herb2rest repository